torch/_dynamo/symbolic_convert.py

```diff
@@ -1090,6 +1090,16 @@ def run(self):
             except Exception as e:
                 if self.exec_recorder:
                     e.exec_record = self.exec_recorder.get_record()  # type: ignore[attr-defined]
+
+                if "Could not guard on data-dependent" in e.msg:
```

Comment:
At this point the guard on data dependent is wrapped by a UserError so we need to resort to string matching.